src/core/thread/mle_router.cpp

```diff
@@ -125,6 +125,11 @@ void MleRouter::SetRouterRoleEnabled(bool aEnabled)
 
         break;
     }
+	
+    if (aEnabled && !mStateUpdateTimer.IsRunning() && IsFullThreadDevice() && (mRole != OT_DEVICE_ROLE_DISABLED))
```

Comment:
Couple thoughts:

1. Should we only start `mStateUpdateTimer` if the device is attached with `Mle::IsAttached()` method?

2. Should we use the `Mle::IsRouterRoleEnabled()` method?

Applying the above would result in something like:
```c++
if (IsRouterRoleEnabled() && IsAttached() && !mStateUpdateTimer.IsRunning())
```

Thoughts?

Reply:
That certainly reads cleaner and covers the same logic.  Nice.